### PR TITLE
Update Default Themes to new Colors/Names

### PR DESCRIPTION
**Pull request by deanwhillier**
#### Summary
This PR replaces the existing `Mattermost`, `Organization`, `Mattermost Dark` & `Windows Dark` theme names and colours with the new `Sapphire`, `Quartz`, `Indigo` & `Onyx` theme names and colours respectively.

Theme changes are applied immediately based on your previously applied theme using a theme mapping object. Custom themes are unaffected. Due to the way this is applied, until you manually change to a new default theme in Account Settings, the old theme will continue to work on linked servers (like MM community servers) that don't have this update applied. If you do manually change to one of the new themes, older linked servers will display the new colours just fine as no changes were made to the theming engine.

Additionally, the theme preview icons have been updated using a dynamic SVG component that gets its colours from each theme directly.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-34861

#### Release Note
```release-note
Updated default theme colors and names of `Mattermost`, `Organization`, `Mattermost Dark` & `Windows Dark` to `Sapphire`, `Quartz`, `Indigo` & `Onyx` theme colors and names respectively.
```

### CODE REVIEW

**Comment by matthewbirtch** [comment]
@deanwhillier it looks like the test server failed. I can review when that's fixed

**Review by esethna — APPROVED** [review]
Sweet! 

The only friction point I see is that the "Organization" theme is mapping to "Quartz" (I think) which is a stark change. But my gut tells me "organization" is probably not a popular theme anyway (plus it's v6.0 and breaking changes are acceptable)

**Review by matthewbirtch — APPROVED** [review]
Looking great @deanwhillier. Thanks!

File: components/post_view/message_attachments/action_button/action_button.test.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ describe('components/post_view/message_attachments/action_button.jsx', () => {
     const baseProps = {
         action: {id: 'action_id_1', name: 'action_name_1', cookie: 'cookie-contents'},
         handleAction: jest.fn(),
-        theme: Constants.THEMES.default as unknown as Theme,
+        theme: Constants.THEMES.sapphire as unknown as Theme,
```
Comment (hmhealey) [inline components/post_view/message_attachments/action_button/action_button.test.tsx:17]: 0/5 on doing it now, but maybe this is a good excuse to move the themes into a new file that can use TypeScript so we stop needing to `as unknown as Theme` every time we use those.

File: components/user_settings/display/user_settings_theme/premade_theme_chooser/premade_theme_chooser.jsx [hunk 2]
```diff
@@ -39,10 +42,19 @@ export default class PremadeThemeChooser extends React.PureComponent {
                             onClick={() => this.props.updateTheme(premadeTheme)}
                         >
                             <label>
-                                <img
-                                    alt={'premade theme ' + k}
-                                    className='img-responsive'
-                                    src={premadeTheme.image}
+                                <ThemeThumbnail
```
Comment (hmhealey) [inline components/user_settings/display/user_settings_theme/premade_theme_chooser/premade_theme_chooser.jsx:45]: Huge thumbs up here. I've wanted this for ages like the crappy version Devin and I made for our one hackathon.

File: components/post_view/message_attachments/action_button/action_button.test.tsx (hunk 1 above)
Comment (hmhealey) [reply components/post_view/message_attachments/action_button/action_button.test.tsx:17]: Actually, maybe we just replace all uses of these with the constants from mattermost-redux?

**Review by hmhealey — APPROVED** [review]

File: components/post_view/message_attachments/action_button/action_button.test.tsx (hunk 1 above)
Comment (deanwhillier) [reply components/post_view/message_attachments/action_button/action_button.test.tsx:17]: Ya, I was wondering about the seemingly duplicated definitions of themes. Probably should have just asked! I can update to only use the ones from mattermost-redux!

**Comment by deanwhillier** [comment]
@matthewbirtch, the colour updates we made to the new themes to support the new Global Header are now included in this PR.

**Comment by jgilliam17** [comment]
Thanks @deanwhillier 
Looks great
Seeing a small discrepancy with the theme preview icons: button, link and new message line colors compared to actual themes on the test server
1. On the test server all themes have blue button background, while preview icon shows only Indigo should have blue, other 3 themes should have green. This also applies to links on the test server, regardless of the theme all are blue.
2. Theme preview icons all have blue "new message line", believe this should match link color as it looks like that on the test server for all themes
Screen Shot 2021-08-02 at 3 26 15 PM

**Review by calebroseland — APPROVED** [review]
🎉

**Comment by deanwhillier** [comment]
Thanks @jgilliam17! As always, excellent work keeping me honest and humble! 😛

I had the variable for the button and new message lines reversed. 😅

**Comment by esethna** [comment]
@amyblais lets please note this in the changelog for cloud AND on-prem when it ships:

The existing Mattermost, Organization, Mattermost Dark and Windows Dark theme names and colours have been updated to the new Sapphire, Quartz, Indigo & Onyx theme names and colours, respectively. Anyone using the existing themes will see slightly modified theme colors after their server is upgraded. The theme variables for the existing Mattermost, Organization, Mattermost Dark and Windows Dark themes will still be accessible in our docs (insert link), so a custom theme can be used with these theme variables if desired. 

cc// @cwarnermm

image

**Pull request merged**